services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -0,0 +1,942 @@
+
+        and: "Update all isls which can be used by protected path"
+        def bandwidth = 100
+        isls[1..-1].each { database.updateIslAvailableBandwidth(it, 90) }
+
+        when: "Create flow with protected path"
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.maximumBandwidth = bandwidth
+        flow.allocateProtectedPath = true
+        flow.ignoreBandwidth = true
+        flowHelper.addFlow(flow)
+
+        then: "Flow is created with protected path"
+        northbound.getFlowPath(flow.id).protectedPath
+
+        and: "Cleanup: delete the flow and restore available bandwidth"
+        flowHelper.deleteFlow(flow.id)
+        isls.each { database.resetIslBandwidth(it) }
+    }
+
+    @Ignore("Test is not valid, the availableBandwidth field is rewritten by each update action")
+    def "Able to enable protected path on a flow in case there is not enough bandwidth for protectedPath and ignoreBandwidth=true"() {
+        given: "Two active neighboring switches"
+        def isls = topology.getIslsForActiveSwitches()
+        def (srcSwitch, dstSwitch) = [isls.first().srcSwitch, isls.first().dstSwitch]
+
+        and: "Update all isls which can be used by protected path"
+        def bandwidth = 100
+        isls[1..-1].each { isl ->
+            database.updateIslAvailableBandwidth(isl, 90)
+            database.updateIslAvailableBandwidth(isl.reversed, 90)
+        }
+
+        when: "Create flow without protected path"
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.maximumBandwidth = bandwidth
+        flow.allocateProtectedPath = false
+        flow.ignoreBandwidth = true
+        flowHelper.addFlow(flow)
+
+        then: "Flow is created without protected path"
+        !northbound.getFlowPath(flow.id).protectedPath
+
+        when: "Update flow: enable protected path(allocateProtectedPath=true)"
+        northbound.updateFlow(flow.id, flow.tap { it.allocateProtectedPath = true })
+
+        then: "Protected path is enabled"
+        northbound.getFlowPath(flow.id).protectedPath
+
+        // TODO I've second thoughts about the following lines
+        northbound.updateFlow(flow.id, flow.tap { it.ignoreBandwidth = false })
+        northbound.updateFlow(flow.id, flow.tap { it.allocateProtectedPath = false })
+        //Should the protected path be enabled?
+//        isls[1..-1].each { isl ->
+//            database.updateIslAvailableBandwidth(isl, 90)
+//            database.updateIslAvailableBandwidth(isl.reversed, 90)
+//        }
+        northbound.updateFlow(flow.id, flow.tap { it.allocateProtectedPath = true })
+
+        and: "Cleanup: delete the flow and restore available bandwidth"
+        flowHelper.deleteFlow(flow.id)
+        isls.each { isl ->
+            database.resetIslBandwidth(isl)
+            database.resetIslBandwidth(isl.reversed)
+        }
+    }
+
+    def "System is able to recalculate protected path when port is down"() {
```

Comment:
It looks like we don't need this test because you already have verifications bellow in other tests.

Reply:
Please, ignore this comment. I had a wrong assumption on it.